ksqldb-execution/src/main/java/io/confluent/ksql/execution/util/ExpressionTypeManager.java

```diff
@@ -146,13 +145,7 @@ public Void visitNotExpression(
 
     @Override
     public Void visitCast(final Cast node, final ExpressionTypeContext expressionTypeContext) {
-      final SqlType sqlType = node.getType().getSqlType();
-      if (!sqlType.supportsCast()) {
-        throw new KsqlFunctionException("Only casts to primitive types or decimals "
-            + "are supported: " + sqlType);
-      }
-
-      expressionTypeContext.setSqlType(sqlType);
+      expressionTypeContext.setSqlType(node.getType().getSqlType());
```

Comment:
Removed this as its not needed.  It's `getCast` in `SqlToJavaVisitor` that determines what casts are supported.  Removing this check just means it fails, with the same error, slightly later in flow.